### DISCUSSION TITLE
Course altitude implemented in wspr db field

### DISCUSSION THREAD

**Discussion by micked99** [mikaelnousiainen/RS41ng#24] [Ideas]
Hi, could it be possible to implements a rough altitude into the wspr db field like something like this, so if its out of line of sight reception it would still be able to have a general idea of float/flight altitude, altitude just an idea, altitudes could easily be altered depending of type of flight, super pressure or latex.

if (alt_meters >= 14600) dbm = 60;
else if (alt_meters >= 14200) dbm = 57;
else if (alt_meters >= 13800) dbm = 53;
else if (alt_meters >= 13400) dbm = 50;
else if (alt_meters >= 13000) dbm = 47;
else if (alt_meters >= 12600) dbm = 43;
else if (alt_meters >= 12200) dbm = 40;
else if (alt_meters >= 11800) dbm = 37;
else if (alt_meters >= 11400) dbm = 33;
else if (alt_meters >= 11000) dbm = 30;
else if (alt_meters >= 10600) dbm = 27;
else if (alt_meters >= 10200) dbm = 23;
else if (alt_meters >= 9800) dbm = 20;
else if (alt_meters >= 9000) dbm = 17;
else if (alt_meters >= 8000) dbm = 13;
else if (alt_meters >= 6000) dbm = 10;
else if (alt_meters >= 4000) dbm = 7;
else if (alt_meters >= 2000) dbm = 3;
else dbm = 0;  

/Mikael

**Comment by mikaelnousiainen** [comment]
Certainly possible, but not sure what the WSPR community would think about this type of "misuse" of the power field :)

**Comment by mikaelnousiainen** [comment]
@micked99 It would be better to have just a simple linear conversion instead of tens of if-statements, something like:

```
power_dbm = altitude_meters * 120 / 15000;
```

WSPR should have 7 bits for power (max value is 127), so the above would give max value of 120 at altitude of 15000. The max altitude could be made configurable.

**Reply by micked99** [reply]
They dont mind us the dbm field for alt, I use it on my currently flying wspr board and have done so for a few years :)

The thing is there is only so many powerlevels and the board floats at a mostly constant altitude that only a 3-4 field are used under a long duration flight (at 400m steps) thats why I think an unlinear method like the one I posted is a better approach
